idea/src/org/jetbrains/kotlin/idea/inspections/RedundantElseInIfInspection.kt

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2010-2018 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license
+ * that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.idea.inspections
+
+import com.intellij.codeInspection.LocalQuickFix
+import com.intellij.codeInspection.ProblemDescriptor
+import com.intellij.codeInspection.ProblemHighlightType
+import com.intellij.codeInspection.ProblemsHolder
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.util.TextRange
+import com.intellij.psi.PsiDocumentManager
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiElementVisitor
+import com.intellij.psi.codeStyle.CodeStyleManager
+import org.jetbrains.kotlin.idea.caches.resolve.analyze
+import org.jetbrains.kotlin.idea.intentions.branchedTransformations.isElseIf
+import org.jetbrains.kotlin.psi.*
+import org.jetbrains.kotlin.psi.psiUtil.endOffset
+import org.jetbrains.kotlin.psi.psiUtil.getStrictParentOfType
+import org.jetbrains.kotlin.psi.psiUtil.startOffset
+import org.jetbrains.kotlin.resolve.BindingContext
+import org.jetbrains.kotlin.types.typeUtil.isNothing
+
+class RedundantElseInIfInspection : AbstractKotlinInspection() {
+    override fun buildVisitor(holder: ProblemsHolder, isOnTheFly: Boolean): PsiElementVisitor =
+        ifExpressionVisitor(fun(ifExpression) {
+            if (ifExpression.isElseIf()) return
+            val elseKeyword = ifExpression.singleElseKeyword(true) ?: return
+            val rangeInElement = elseKeyword.textRange?.shiftRight(-ifExpression.startOffset) ?: return
+            holder.registerProblem(
+                holder.manager.createProblemDescriptor(
+                    ifExpression,
+                    rangeInElement,
+                    "Redundant 'else'",
+                    ProblemHighlightType.LIKE_UNUSED_SYMBOL,
```

Comment:
Don't report it as unused symbol, otherwise it looks like the whole `else` block is redundant which is not true.